### PR TITLE
update pythonSrc/ext to not use parse-1.8.4

### PR DESCRIPTION
**Pull request by dkrieger-pvtl**
This PR reverts the expedient inclusion of parse-1.8.4.tar.gz in pythonSrc/ext in favor of the better solution that unvendors Behave (https://github.com/greenplum-db/gpdb/pull/8141) that has been merged.

This PR and its pythonSrc/ext companion PR(https://github.com/greenplum-db/pythonsrc-ext/pull/7) effectively reverts https://github.com/greenplum-db/gpdb/pull/8138.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [submodule "gpMgmt/bin/pythonSrc/ext"]
 	path = gpMgmt/bin/pythonSrc/ext
 	url = https://github.com/greenplum-db/pythonsrc-ext.git
-	branch = master
+	branch = 167291492_behave_misbehaves_ext_revert
```
Comment (BaiShaoqi) [inline .gitmodules]: After the https://github.com/greenplum-db/pythonsrc-ext/pull/7 merged, this change do not need right?

File: concourse/pipelines/templates/gpdb-tpl.yml [hunk 2]
```diff
@@ -1522,6 +1522,7 @@ jobs:
         params:
           submodules:
             - gpMgmt/bin/pythonSrc/ext
+          submodule_remote: true
```
Comment (kalensk) [inline concourse/pipelines/templates/gpdb-tpl.yml]: Just curious if we _really_ want to be introducing both the `submodule_remote: true`,  and changing the .gitmodules branch from `master` to `167291492_behave_misbehaves_ext_revert`?

Or do we _not_ want those changes and simply add the corresponding commit to gpMgmt/bin/pythonSrc/ext in the gpdb repo as shown in the last file changed below?

File: .gitmodules (hunk 1 above)
Comment (dkrieger-pvtl) [reply .gitmodules]: I should have been clearer; only the pythonSrc/ext remove of parse-1.8.4.tar.gz and updating its sha in gpdb will be done; this change is only for developmental testing.

```
(revert before merging) use updated pythonSrc/ext without parse-1.8.4.

In order to validate changes to pythonSrc/ext, we need to use a branch.
But that requires these (dev-only) changes.  This commit will be
reverted before merging.
```

File: concourse/pipelines/templates/gpdb-tpl.yml (hunk 2 above)
Comment (dkrieger-pvtl) [reply concourse/pipelines/templates/gpdb-tpl.yml]: I should have been clearer; only the pythonSrc/ext remove of parse-1.8.4.tar.gz and updating its sha in gpdb will be done; this change is only for developmental testing.

```
(revert before merging) use updated pythonSrc/ext without parse-1.8.4.

In order to validate changes to pythonSrc/ext, we need to use a branch.
But that requires these (dev-only) changes.  This commit will be
reverted before merging.
```

**Pull request merged**